xbmc/utils/EGLUtils.cpp

```diff
@@ -144,42 +145,68 @@ bool CEGLContextUtils::InitializeDisplay(EGLint renderableType, EGLint rendering
     return false;
   }
 
+  if (!ChooseConfig(renderableType, visualId))
+    return false;
+
+  return true;
+}
+
+bool CEGLContextUtils::ChooseConfig(EGLint renderableType, EGLint visualId)
+{
+  EGLint numConfigs{0};
+  EGLint numMatched{0};
+
   EGLint surfaceType = EGL_WINDOW_BIT;
   // for the non-trivial dirty region modes, we need the EGL buffer to be preserved across updates
   if (g_advancedSettings.m_guiAlgorithmDirtyRegions == DIRTYREGION_SOLVER_COST_REDUCTION ||
       g_advancedSettings.m_guiAlgorithmDirtyRegions == DIRTYREGION_SOLVER_UNION)
     surfaceType |= EGL_SWAP_BEHAVIOR_PRESERVED_BIT;
 
-  EGLint attribs[] =
-  {
-    EGL_RED_SIZE, 8,
-    EGL_GREEN_SIZE, 8,
-    EGL_BLUE_SIZE, 8,
-    EGL_ALPHA_SIZE, 8,
-    EGL_DEPTH_SIZE, 16,
-    EGL_STENCIL_SIZE, 0,
-    EGL_SAMPLE_BUFFERS, 0,
-    EGL_SAMPLES, 0,
-    EGL_SURFACE_TYPE, surfaceType,
-    EGL_RENDERABLE_TYPE, renderableType,
-    EGL_NONE
-  };
-
-  EGLint neglconfigs = 0;
-  if (eglChooseConfig(m_eglDisplay, attribs, &m_eglConfig, 1, &neglconfigs) != EGL_TRUE)
+  CEGLAttributes<10> attribs;
+  attribs.Add({{EGL_RED_SIZE, 8},
+               {EGL_GREEN_SIZE, 8},
+               {EGL_BLUE_SIZE, 8},
+               {EGL_ALPHA_SIZE, 8},
+               {EGL_DEPTH_SIZE, 16},
```

Comment:
EGL_DEPTH_SIZE=16 is problematic with some implementations, e.g. lima only supports '8', do we really need 16 bits for the z position?
In the past I just patched out the line in offending apps (e.g. kmscube) and that seemed to work fine.

Reply:
@koenkooi That is something I can explore but I do net yet understand the implications of changing this value. I have tried to google to see how to determine what this value should be but so far have come up short.

Reply:
Running your small test app I get only 0 or 24 as depth size on lima:
```
matched configs: 12

config: 0
  EGL_RED_SIZE: 8 (0x0008)
  EGL_GREEN_SIZE: 8 (0x0008)
  EGL_BLUE_SIZE: 8 (0x0008)
  EGL_ALPHA_SIZE: 0 (0x0000)
  EGL_DEPTH_SIZE: 0 (0x0000)
  EGL_NATIVE_VISUAL_ID: 875713112 (0x34325258)
  EGL_SAMPLES: 0 (0x0000)

config: 1
  EGL_RED_SIZE: 8 (0x0008)
  EGL_GREEN_SIZE: 8 (0x0008)
  EGL_BLUE_SIZE: 8 (0x0008)
  EGL_ALPHA_SIZE: 0 (0x0000)
  EGL_DEPTH_SIZE: 24 (0x0018)
  EGL_NATIVE_VISUAL_ID: 875713112 (0x34325258)
  EGL_SAMPLES: 0 (0x0000)

config: 2
  EGL_RED_SIZE: 8 (0x0008)
  EGL_GREEN_SIZE: 8 (0x0008)
  EGL_BLUE_SIZE: 8 (0x0008)
  EGL_ALPHA_SIZE: 0 (0x0000)
  EGL_DEPTH_SIZE: 24 (0x0018)
  EGL_NATIVE_VISUAL_ID: 875713112 (0x34325258)
  EGL_SAMPLES: 0 (0x0000)

config: 3
  EGL_RED_SIZE: 8 (0x0008)
  EGL_GREEN_SIZE: 8 (0x0008)
  EGL_BLUE_SIZE: 8 (0x0008)
  EGL_ALPHA_SIZE: 0 (0x0000)
  EGL_DEPTH_SIZE: 0 (0x0000)
  EGL_NATIVE_VISUAL_ID: 875713112 (0x34325258)
  EGL_SAMPLES: 4 (0x0004)

config: 4
  EGL_RED_SIZE: 8 (0x0008)
  EGL_GREEN_SIZE: 8 (0x0008)
  EGL_BLUE_SIZE: 8 (0x0008)
  EGL_ALPHA_SIZE: 0 (0x0000)
  EGL_DEPTH_SIZE: 24 (0x0018)
  EGL_NATIVE_VISUAL_ID: 875713112 (0x34325258)
  EGL_SAMPLES: 4 (0x0004)

config: 5
  EGL_RED_SIZE: 8 (0x0008)
  EGL_GREEN_SIZE: 8 (0x0008)
  EGL_BLUE_SIZE: 8 (0x0008)
  EGL_ALPHA_SIZE: 0 (0x0000)
  EGL_DEPTH_SIZE: 24 (0x0018)
  EGL_NATIVE_VISUAL_ID: 875713112 (0x34325258)
  EGL_SAMPLES: 4 (0x0004)

config: 6
  EGL_RED_SIZE: 8 (0x0008)
  EGL_GREEN_SIZE: 8 (0x0008)
  EGL_BLUE_SIZE: 8 (0x0008)
  EGL_ALPHA_SIZE: 8 (0x0008)
  EGL_DEPTH_SIZE: 0 (0x0000)
  EGL_NATIVE_VISUAL_ID: 875713089 (0x34325241)
  EGL_SAMPLES: 0 (0x0000)

config: 7
  EGL_RED_SIZE: 8 (0x0008)
  EGL_GREEN_SIZE: 8 (0x0008)
  EGL_BLUE_SIZE: 8 (0x0008)
  EGL_ALPHA_SIZE: 8 (0x0008)
  EGL_DEPTH_SIZE: 24 (0x0018)
  EGL_NATIVE_VISUAL_ID: 875713089 (0x34325241)
  EGL_SAMPLES: 0 (0x0000)

config: 8
  EGL_RED_SIZE: 8 (0x0008)
  EGL_GREEN_SIZE: 8 (0x0008)
  EGL_BLUE_SIZE: 8 (0x0008)
  EGL_ALPHA_SIZE: 8 (0x0008)
  EGL_DEPTH_SIZE: 24 (0x0018)
  EGL_NATIVE_VISUAL_ID: 875713089 (0x34325241)
  EGL_SAMPLES: 0 (0x0000)

config: 9
  EGL_RED_SIZE: 8 (0x0008)
  EGL_GREEN_SIZE: 8 (0x0008)
  EGL_BLUE_SIZE: 8 (0x0008)
  EGL_ALPHA_SIZE: 8 (0x0008)
  EGL_DEPTH_SIZE: 0 (0x0000)
  EGL_NATIVE_VISUAL_ID: 875713089 (0x34325241)
  EGL_SAMPLES: 4 (0x0004)

config: 10
  EGL_RED_SIZE: 8 (0x0008)
  EGL_GREEN_SIZE: 8 (0x0008)
  EGL_BLUE_SIZE: 8 (0x0008)
  EGL_ALPHA_SIZE: 8 (0x0008)
  EGL_DEPTH_SIZE: 24 (0x0018)
  EGL_NATIVE_VISUAL_ID: 875713089 (0x34325241)
  EGL_SAMPLES: 4 (0x0004)

config: 11
  EGL_RED_SIZE: 8 (0x0008)
  EGL_GREEN_SIZE: 8 (0x0008)
  EGL_BLUE_SIZE: 8 (0x0008)
  EGL_ALPHA_SIZE: 8 (0x0008)
  EGL_DEPTH_SIZE: 24 (0x0018)
  EGL_NATIVE_VISUAL_ID: 875713089 (0x34325241)
  EGL_SAMPLES: 4 (0x0004)
```